libraries/fc/include/fc/io/cfile.hpp

```diff
@@ -79,9 +84,17 @@ class cfile {
    }
 
    void skip( long loc) {
-      if( 0 != fseek( _file.get(), loc, SEEK_CUR ) ) {
-         throw std::ios_base::failure( "cfile: " + _file_path.generic_string() +
-                                       " unable to SEEK_CUR to: " + std::to_string(loc) );
+      if( loc < 0 ) {
+         if( 0 != fseek( _file.get(), loc, SEEK_CUR ) ) {
+            throw std::ios_base::failure( "cfile: " + _file_path.generic_string() +
+                                          " unable to SEEK_CUR to: " + std::to_string(loc) );
+         }
+      }
+      else while( loc > 0 ) {
+         char bit_bucket[4096];
+         auto read_max = std::min( long(sizeof(bit_bucket)), loc );
+         read( bit_bucket, read_max );
+         loc -= read_max;
```

Comment:
Does this mean `fseek` just doesn't work? Why is this needed?

Reply:
Seeking beyond the end of the file isn't an error, you need to actually read from it to see an EOF error. I thought maybe just do the `fseek` and call `eof()` but the problem with that is that if someone deliberately tries to `skip()` to the end _because they then want to append at the end_, there shouldn't be an error in the case.

Reply:
Can't you also seek past the end of the file and start writing as well?  Anyway, seeking to the end should definitely not be an error.

Reply:
Instead of the bit bucket, we could do something like `seek()` that probes the file size instead. Without knowing average size of `skip()` it's hard to know which will be more efficient (I expect each time probing the file size is a syscall).

Reply:
Why do you need to detect whether the file position passed the end of file exactly at seek() and skip()? In normal C programing, it's detected at the next read() operation and it's generally good enough. In detecting at seek()/skip(), you are adding a lot of overhead for normal usage of these two operations.

Reply:
okay, if that's what you all want, I will remove it